packages/web-components/src/components/fade-in-out/fade-in-out.ts

```diff
@@ -0,0 +1,208 @@
+   *
+   * @private
+   */
+
+  private _getViewportMargin() {
+    return `-${((document.documentElement.clientHeight * this._colSpan) / breakpoints.max.columns).toString()}px 0px`;
+  }
+
+  /**
+   * Component that handles fade transition for selected elements.
+   *
+   * @example
+   * import '@carbon/ibmdotcom-styles/scss/internal/scroll-into-view/_scroll-into-view.scss';
+   * import '@carbon/ibmdotcom-web-components/es/components/fade-in-out/fade-in-out.js';
+   *
+   * As an example, the function can be called to target all instances of the
+   * elements in a list:
+   *
+   * const list = ['.bx--content-block', 'bx--content-group'];
+   *
+   * For default values of 400ms and 'one and done' play:
+   * <dds-fade-in-out element-list="${JSON.stringify(list)}"/>
+   *
+   * With 'continuous play' option:
+   * <dds-fade-in-out element-list="${JSON.stringify(elementList)}" iterations="true"/>
+   *
+   * For custom delay time, set within targeted class in the application's CSS code as such:
+   *
+   * .bx--content-block {
+   *   --#{$dds-prefix}--scroll-into-view-delay: 250ms;
+   * }
+   *
+   * @param {boolean} create to create new observer instances
+   */
+
+  private _cleanAndCreateObservers({ create }: { create?: boolean } = {}) {
+    if (this._rootObserver) {
+      this._rootObserver.disconnect();
+      this._innerObserver.disconnect();
+      this._resizeObserver.disconnect();
+      this._rootObserver = null;
+      this._innerObserver = null;
+      this._resizeObserver = null;
+    }
+
+    if (create) {
+      this._rootObserver = new IntersectionObserver(this.handleExit);
+      this._innerObserver = new IntersectionObserver(this.handleEntrance.bind(this), this._options);
+      this._resizeObserver = new ResizeObserver(this.handleResize.bind(this));
+
+      this.elementList?.forEach(selector => {
+        const elements = document.querySelectorAll(selector);
+        elements.forEach(e => {
+          this._rootObserver.observe(e);
+          this._innerObserver.observe(e);
+        });
+      });
+      this._resizeObserver.observe(document.documentElement);
+    }
+  }
+
+  /**
+   * Handler to add recalculated rootMargin to observer.
+   *
+   * @private
+   *
+   */
+
+  private handleResize() {
+    this._options.rootMargin = this._getViewportMargin();
+    this._innerObserver = new IntersectionObserver(this.handleEntrance.bind(this), this._options);
+  }
+
+  /**
+   * Handler to add fade animation to element
+   *
+   * @param {*} entries observed elements
+   * @private
+   *
+   */
+
+  private handleEntrance(entries) {
+    entries.forEach(entry => {
+      if (entry.intersectionRatio > 0) {
+        entry.target.classList.remove('bx--fade-out');
+        entry.target.classList.add('bx--fade-in');
+        if (!this._iterations) {
+          this._rootObserver.unobserve(entry.target);
+          this._innerObserver.unobserve(entry.target);
+        }
+      }
+    });
+  }
```

Comment:
* Make sure private methods are prefixed with `_`.
* Make sure we have type annotations to method args.
* Usage of arrow function avoids the need for `.bind()`.
* Style nit: `<dds-table-of-contents>` uses `records` arg for `IntersectionObserver` callback. I think it's good to align the naming strategy.

```suggestion
  private _handleEntrance = (records: IntersectionObserverEntry[]) => {
    records.forEach({ intersectionRatio, target } => {
      if (intersectionRatio > 0) {
        target.classList.remove('bx--fade-out');
        target.classList.add('bx--fade-in');
        if (!this._iterations) {
          this._rootObserver.unobserve(target);
          this._innerObserver.unobserve(target);
        }
      }
    });
  };
```